src/Chapter3.hs

```diff
@@ -971,6 +1098,22 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday deriving (Show, Eq, Enum, Bounded)
+
+isWeekend :: Weekday -> Bool
+isWeekend weekday = weekday == Sunday || weekday == Saturday
+
+-- Bounded gives the minBound ( Sunday in this case) and maxBound (Saturday)
+-- Enum gives us succ to move to next in enum
+nextDay :: Weekday -> Weekday
+nextDay weekday
+  | weekday == maxBound = Monday
+  | otherwise = succ weekday
```

Comment:
That is right!

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂